functional-tests/utils.py

```diff
@@ -102,6 +102,19 @@ def wait_until_with_value(
     raise AssertionError(error_with)
 
 
+def wait_until_checkpoint(seqrpc, timeout: int = 5, step: float = 2):
+    """Waits until the current checkpoint index increases."""
+    first_ckpt_idx = seqrpc.strata_getLatestCheckpointIndex()
+
+    def _f():
+        status = seqrpc.strata_syncStatus()
+        print("waiting for epoch, at chain tip slot", status)
+        cur_ckpt_index = seqrpc.strata_getLatestCheckpointIndex()
+        return cur_ckpt_idx > first_ckpt_idx
+
+    wait_until_with_value(_f, lambda v: v, "Epoch never finalized", timeout, step)
```

Comment:
The caller doesn't seem to use the returned value, so could just be simplified with `wait_until`.
```python
wait_until(_f, "Error never vinalizesd", timeout, step)
```

Reply:
Oh good point, I didn't realize what I was doing.